### DISCUSSION TITLE
Is toast action/close intentionally designed to not resume?

### DISCUSSION THREAD

**Discussion by gdfreitasdev** [radix-ui/primitives#2221] [Help] [answered]
I was wondering why the components `<Toast.Action>|<Toast.Close>` does not resume the paused state when clicked. 

It maintains the state of paused for new upcoming toasts which imo might not be the ideal behavior as it will require user interaction to resume toasts automatically again. 

What do you think? Should this be filed as a bug?

To reproduce we can use official docs: https://www.radix-ui.com/docs/primitives/components/toast#close
1. Click on "Add to calendar"
2. Click on "Undo" inside toast
3. Click on "Add to calendar" and notice that it won’t disappear automatically anymore

This can also be reproduced on other libraries/design systems that uses Radix internally
- https://ui.shadcn.com/docs/components/toast
- https://vercel.com/design/toast#action-cancel

**Comment by jvzaniolo** [comment, marked as answer]
@gdfreitas This issue is happening because when you click the dismiss/close button, it changes the `context.isClosePausedRef` current value to `true`, so when you create a new Toast, it does not reset its value to `false` and this verification doesn't trigger the `startTimer` function to dismiss the Toast automatically.

https://github.com/radix-ui/primitives/blob/eca6babd188df465f64f23f3584738b85dba610e/packages/react/toast/src/Toast.tsx#L538

I opened a PR to fix this and another similar issue with resetting the `ref` current values.

- #2236

**Reply by gdfreitasdev** [reply]
That's what I was suspecting, thanks for the response @jvzaniolo . I'll subscribe to your PR.

**Reply by adolfoscheidt** [reply]
I had the same issue, thanks too! Subscribing to the PR as well
